src/modules/ROOT/pages/rfc/record-value-expr/index.adoc

```diff
@@ -0,0 +1,138 @@
+
+== Formula field
+
+Formula fields allow dynamic calculation of record values from constants and other values.
+Only constants and values from current objects are supported.
+
+.Example expression for formula field that calculates price with tax:
+[source]
+----
+price * 1.25
+----
+
+This expression assumes there is another field named `price` in record values and multiplies that field with 1.25.
+
+.Example expression for formula field that concatinates 2 string fields
+[source]
+----
+trim(firstName + " " + lastName)
+----
+
+Expression assumes 2 other fields (firstName and lastName) and concatenates them with space between.
+Finally, we trim the result to ensure there result is not prefixed or suffixed with a space in case any of the two fields are empty.
+
+_Later upgrades will allow calculation from referenced records with dependency tracking and cascading updates._
+
+#TBD# How to handle formula change on existing data.
+
+== Sanitizers
+
+Sanitizers aid in record value processing and *cannot raise errors* but *can modify value*.
+Each module field can have one or more defined expression with accompanying description or explanation.
+
+.Example of length control
+[source]
+----
+length(title) > 5 ? substr(title, 0, 5) + "..." : title
+----
+.dfdfd
+
+
+Expressions are ran *before* built-in sanitizers. Built-in sanitizers cannot be disabled.
+When multiple sanitization expressions are defined they are all ran sequentially in defined order.
+
+Corteza logs all modifications to sanitizators.
+
+#TBD Explicit logging could be enabled to catch all sanitized data (where input value differs from final one) for a particular expression"
+
+== Validators
+
+Validators aid in record value processing, *can raise errors* in case of invalid values but *cannot change values*.
+Each module field can have one or more validation groups that consist of validation expression and  error message.
+Built-in field validators can be disabled.
+
+.Example of length control
+[source]
+----
+length(title) < 5
+----
+
+When multiple validation expressions are defined they are all ran sequentially in defined order.
+Custom validators are ran *after* or *instead* of built-in field-type validators.
+Note that validation can produce multiple errors from different fields but each value is validated only until first error (or success).
+
+Corteza logs all modifications to validators.
+
+#TBD Explicit logging could be enabled to catch and log errors for a expression#
+
+== Formatters
+
+When record values are prepared for output Corteza formats each one according to field type and custom formatters.
+Formatters *can reformat value* before they are outputed but *cannot raise errors or change them*.
+Each module field can have one or more formatter expression.
+
+Custom formatters are executed *after* or *instead* of built-on formatters.
+
+
+.Example of custom formatting
+[source]
+----
+capitalize(title)
+----
+
+== Expression functions (wip)
```

Comment:
Is chaining possible?

Reply:
`foo(bar(baz()))` yes, `foo().bar().baz()` probably not.